src/entry_point.py

```diff
@@ -136,9 +144,66 @@ def __call__(self, parser, namespace, values, option_string=None):
         "in the environment specified by --prefix",
     )
 
+    subcommands = p.add_subparsers(dest="command")
+    uninstall_subcommand = subcommands.add_parser(
+        "uninstall",
+        description="Uninstalls a conda directory and all environments inside the directory.",
+    )
+    uninstall_subcommand.add_argument(
+        "--prefix",
+        action="store",
+        required=True,
+        help="Path to the conda directory to uninstall.",
+    )
+    uninstall_subcommand.add_argument(
+        "--remove-caches",
+        action="store_true",
+        required=False,
+        help=(
+            "Removes the notices cache and runs conda --clean --all to clean package caches"
+            " outside the installation directory."
+            " This is especially useful when pkgs_dirs is set in a .condarc file."
+            " Not recommended with multiple conda installations when softlinks are enabled."
+        ),
+    )
+    uninstall_subcommand.add_argument(
+        "--remove-config-files",
+        choices=["user", "system", "all"],
+        default=None,
+        required=False,
+        help=(
+            "Removes all .condarc files."
+            " `user` removes the files inside the current user's"
+            " home directory and `system` removes all files outside of that directory."
+            " Not recommended when multiple conda installations are on the system"
+            " or when running on an environments directory."
+        ),
+    )
+    uninstall_subcommand.add_argument(
+        "--remove-user-data",
+        action="store_true",
+        required=False,
+        help=(
+            "Removes the ~/.conda directory."
+            " Not recommended when multiple conda installations are on the system"
+            " or when running on an environments directory."
+        ),
+    )
+
     args, args_unknown = p.parse_known_args()
 
-    args.prefix = os.path.abspath(args.prefix)
+    if args.command != "uninstall":
+        group_args = getattr(g, "_group_actions")
+        if all(getattr(args, arg.dest) is None for arg in group_args):
+            required_args = [arg.option_strings[0] for arg in group_args]
+            raise argparse.ArgumentError(
+                f"one of the following arguments are required: {'/'.join(required_args)}"
+            )
+
+    if args.prefix is None:
+        raise argparse.ArgumentError("the following arguments are required: --prefix")
+
+    args.prefix = os.path.abspath(os.path.expanduser(os.path.expandvars(args.prefix)))
     args.root_prefix = os.path.abspath(os.environ.get("CONDA_ROOT_PREFIX", args.prefix))
```

Comment:
I guess we don't need to expand {user,vars} `root_prefix` too, or do we?

Reply:
I'm not sure how `os.environ` expands variables here. I'd assume that they are already expanded?